sphinx_gallery/gen_rst.py

```diff
@@ -498,11 +498,8 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     ref_fname = example_file.replace(os.path.sep, '_')
     example_rst = """\n\n.. _sphx_glr_{0}:\n\n""".format(ref_fname)
 
-    if not fname.startswith('plot'):
-        convert_func = dict(code=codestr2rst, text=text2string)
-        for blabel, bcontent in script_blocks:
-            example_rst += convert_func[blabel](bcontent) + '\n'
-    else:
+    filename_pattern = gallery_conf.get('filename_pattern', 'plot_*')
+    if fnmatch.fnmatch(fname, filename_pattern):
```

Comment:
Can you describe a usecase where a complex rule is justified? In my honest opinion, the target users were _not_ advanced users but those who wanted to contribute only a section of the documentation and did not want to build the entire documentation.